### DISCUSSION TITLE
Brand new Wyze V3, RTSP not showing up

### DISCUSSION THREAD

**Discussion by cswanson2** [gtxaspec/wz_mini_hacks#497] [Q&A]
Bought two brand new V3 cameras
- firmware: 4.36.8.15
- followed the exact setup and install steps in the docs
- formatted the drive
- copied the files
- inserted into off camera and plugged it in

It said
- "wz SD Mini first boot" or something close to that
- "extracting swap, please wait"

eventually turns blue
Camera works fine
But no RTSP option in Advanced settings

Several videos online mention booting with the setup button held down - that also doesn't make anything different happen.
Tried two different microSD cards - both boot with the proper messages
Tried resetting the camera to stock and trying again - nothing different.

So this is so far not working.

Any help greatly appreciated.

**Comment by Bob-AOL** [comment]
The camera requires the Wyze RTSP Beta 4.61.0.3 firmare to use the app for activating the onboard RTSP server of the Wyze Beta firmware. 

That's choice #1 for an RTSP solution via the Wyze app itself. Even if the wz_mini firmware is running with RTSP activated, the Wyze RTSP server will prevail, though most of the optional RTSP settings available with wz_mini will still function as intended. 

If using a standard Wyze OEM 4.36.x.x  firmware version, the wz_mini RTSP functionality needs to be activated according to the README instructions. It is strongly encouraged to upgrade the 4.36 version to 9.139 for the best stability and latest features if intending to still use the Wyze features. But even with the wz_mini RTSP option enabled, it will not appear as an option in the Wyze app.

**Comment by bjs-pdx** [comment]
As per @Bob-AOL, read this section of the readme file for RTSP setup instructions:

https://github.com/gtxaspec/wz_mini_hacks#rtsp-streaming

**Comment by cinsun42** [comment]
I don't mean to hijack this thread, but I'm having a nearly identical issue and hope maybe we can both sort this out.

I have two new wyze cam v3's that I can't get RTSP streaming to work on. I have confirmed connectivity to each camera after a successful install of wz_mini_hacks, both by accessing the camera through ssh and the webserver.

I have flashed both 4.36.9.131 AND 4.36.9.139, neither of which have produced a working RTSP stream. I am testing the stream both through VLC on a linux laptop and tinycam through an android phone.

I've read through the readme file countless times, following the instructions to enable RTSP, and tried every combination of variables I can imagine to get either the high res, low res, or both streams to function. I've not gotten anything to work at this point - I simply do not get any data transfer or successful connection to the camera by way of RTSP. At this point I feel like I've exhausted all my options and would greatly appreciate any thoughts or feedback on how to proceed.

I'd like to stick with using wz mini hacks to use the camera in self hosted mode and enable NFS, with use of the RTSP firmware and wyze bridge docker app being significantly less attractive options. Thanks in advance for any help!

**Comment by Bob-AOL** [comment]
Hijack away. Let's find your problem.

To be clear, the Wyze OEM firmware does not produce the stream nor do anything else not natively inherent in its behavior..., though its  user/pass info is what is required for the wz_mini RTSP stream. It otherwize simply 'allows' wz_mini to co-exist through a modified boot process introduced via the SD card. 

An exception to RTSP functionality _does_ occur when the Wyze RTSP Beta 4.61.0.1~3 firmwware is installed in the camera as explained previously. That firmware can be used quite succesfully in combination with other wz_mini features, presumably including self-hosted mode, though I'm not aware of anyone trying it untethered from the Wyze infrastructure. (No reason it shouldn't immediately celebrate its newfound independence!)
 
The first thing I'd do is confirm your wz_mini.conf edits are actually be written out and saved if doing them live online by either of the methods you described.  There are several moving parts with either method that may or not play nice for any number of reasons -- even for a seasoned Linux user. 

Editing with Windows is actually more straightforward and efficient in my humble opinion. The caveat there can be Notepad if using it as your plain text editor. Many will say it won't work, but it works just fine when editing a Unix LF based document provided the document remains it its original LF format and is not copied, edited and saved as a Windows text file using the Windows CRLF convention. Unix (LF) will be displayed at the bottom of the document if everything is cool. A fancy text editor of any flavor won't work. Plain text editing using the Unix LF convention is required.

Check your wz_mini.conf file. The active wz_mini folder should contain any previous versions of wz_mini.conf. If there's only one, it is the original one in either its default form or a pre-edited offline version if you edited it before committing it to the SD card -- which is advisable for basic initial settings (with the exception of the USB/POE function). 

If you have made changes to the wz_mini.conf file other than enabling RTSP functionality, start over from scratch with the clean file in the distro SD_ ROOT folder. Keep it as simple as possible only enabling the web server and these RTSP settings for starters. Nothing else.

#####VIDEO STREAM#####
RTSP_LOGIN="youruser"
RTSP_PASSWORD="youruserpass"
RTSP_PORT="554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="false"
RTSP_HI_RES_FPS="20"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"

Stick with port 554 just because.

The MRL will be rtsp://IP.a.ddr.ess:554/unicast  You should be prompted for the login credentials by whatever platform you're using to display the stream and not need to include them in the MRL. And just to be clear here:: _the credentials are the wifi network info for the camera itself, not something new that you create just for the stream.

Start over from here and see where it goes.

Good luck.

**Reply by cinsun42** [reply]
Thanks @Bob-AOL for the guidance, I'm still working at it.

Per your suggestions, I've started with a new config file and modified only the RTSP and webserver sections. Still no dice - when attempting to open the network stream in VLC I'm still receiving the same error "Your input can't be opened". I did observe something interesting when attempting to access the stream with the camera off, VLC seemed to spend a few seconds searching for the stream before throwing an error, whereas with the camera powered VLC would immediately throw an error. This tells me VLC can access the camera through the network, but is unable to start the stream - this makes sense as I'm still able to access all other functionality of the camera through my network (ssh, webserver, etc as noted previously)

I did copy and use the exact same settings for RTSP that you provided. I'm not exactly clear what you mean by the credentials being the "wifi network info for the camera itself". I thought the credentials defined in the config file are whats used to access the RTSP stream, so in that way somewhat arbitrary. Are you saying these credentials are defined elsewhere and need to be included in the config? For testing purposes, I did simply disable stream authentication to see if it had any effect and it did not.

**Comment by bjs-pdx** [comment]
Try disabling authentication for rtsp:

RTSP_AUTH_DISABLE="false"

Also try enabling audio.  I seem to recall some issues if it is disabled.

**Comment by Bob-AOL** [comment]
> Are you saying these credentials are defined elsewhere and need to be included in the config?

Yes. The credentials need to be the username and password used when initially setting up your camera for your wifi network.

**Comment by bjs-pdx** [comment]
These are settings I just tested that work for me:
```
#####VIDEO STREAM#####
RTSP_LOGIN="admin"
RTSP_PASSWORD=""
RTSP_PORT="8554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_FPS="20"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"
RTSP_LOW_RES_FPS=""
RTSP_LOW_RES_MAX_BITRATE=""
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""

RTSP_AUTH_DISABLE="true"
```

**Reply by cinsun42** [reply]
Just tried the settings above, unfortunately its not working for me.

I'm starting to wonder if this is network related, despite my ability to access the camera. I am using pfsense and have found some threads discussing issues with RTSP streams crossing VLANs. I do have separate VLANs set up on my network, but this access issue persists for me regardless of whether I'm crossing VLANs or staying within the same subnet - so I don't know if its tied to that directly. I'm trying to perform some packet capture to find ANYTHING meaningful...

**Reply by cinsun42** [reply]
Ok, so I got it working!

For whatever reason, it seems VLC just did not want to play nice. I gave up and went back to messing with tinycam. I realized that by adding the cameras as "Wyze" branded cameras in the app, it would change the formatting of the RTSP link (probably to what the wyze "official" rtsp link would be). I re-added it as a "generic" camera to ensure the rtsp link was not modified and BINGO BANGO BONGO it works!! Thanks @Bob-AOL and @bjs-pdx for the support!

Follow up question, with these wyze cams is there a general preference between TCP vs UDP?

**Reply by Bob-AOL** [reply]
I'd stick with 'auto' if available! LOL

Congrats.

**Comment by Bob-AOL** [comment]
You forgot to mention those little details. LOL

Get all the crap out of the way so the camera can get directly to the device running VLC via wifi (or the least convoluted path) and see what happens.
